src/smolagents/models.py

```diff
@@ -1161,6 +1170,136 @@ def create_client(self):
+        boto3_client (`boto3.client`, *optional*):
+            A custom boto3 client for AWS interactions. If not provided, a default client will be created.
+        inference_config (`dict`, *optional*):
+            Configuration parameters for model inference, such as temperature, max tokens, and other model-specific settings.
+        guardrail_config (`dict`, *optional*):
+            Configuration for model guardrails to control and moderate model behavior and outputs.
+        api_kwargs (`dict`, *optional*):
+            Additional keyword arguments to be passed to boto3 bedrock-runtime API requests.
+        custom_role_conversions (`dict[str, str]`, *optional*):
+            Custom role conversion mapping to convert message roles in others.
+            Useful for specific models that do not support specific message roles like "system".
+        flatten_messages_as_text (`bool`, default `False`):
+            Whether to flatten messages as text.
+
+    Example:
+        >>> bedrock_model = AmazonBedrockServerModel(
+        ...     model_id='us.amazon.nova-pro-v1:0',
+        ...     inference_config={'temperature': 0.7, 'max_tokens': 3000}
+        ... )
+    """
+
+    def __init__(
+        self,
+        model_id: str,
+        boto3_client=None,
+        inference_config: dict[str, Any] | None = None,
+        guardrail_config: dict[str, Any] | None = None,
+        bedrock_api_kwargs: dict[str, Any] | None = None,
+        custom_role_conversions: dict[str, str] | None = None,
+        flatten_messages_as_text: bool = False,
+        **kwargs,
+    ):
+        self.model_id = model_id
+        self.client = boto3_client or self.create_client()
+        self.inference_config = inference_config
+        self.guardrail_config = guardrail_config
+        self.bedrock_api_kwargs = bedrock_api_kwargs
+        self.custom_role_conversions = custom_role_conversions
+
+        super().__init__(
+            model_id=model_id,
+            custom_role_conversions=custom_role_conversions,
+            flatten_messages_as_text=flatten_messages_as_text,
+            **kwargs,
+        )
+
+    def create_client(self):
+        try:
+            import boto3
+        except ModuleNotFoundError as e:
+            raise ModuleNotFoundError(
+                "Please install 'aws_sdk' extra to use AmazonBedrockServerModel: `pip install 'smolagents[aws_sdk]'`"
+            ) from e
+
+        return boto3.client("bedrock-runtime")
+
+    def __call__(
+        self,
+        messages: List[Dict[str, str]],
+        tools_to_call_from: Optional[List[Tool]] = None,
+        **kwargs,
+    ) -> ChatMessage:
+        # smolagents inject automatic parameters like end_stop, Observation and others
+        # The Bedrock API only supports the fields explicitly defined here.
+        # To ensure compliance, `kwargs` is set to empty, and customers must provide additional parameters using `bedrock_api_kwargs`.
+        kwargs = {}
```

Comment:
I think this should be moved to your `_prepare_completion_kwargs` overriding (see below).

Reply:
Yeah!